setup.py

```diff
@@ -596,7 +596,7 @@ def cython_extensions_and_necessity():
     packages=list(PACKAGES),
     package_dir=PACKAGE_DIRECTORIES,
     package_data=PACKAGE_DATA,
-    python_requires=">=3.8",
+    python_requires=f">={min_python_version}",
```

Comment:
We can just use `python_version.MIN_PYTHON_VERSION` directly here, same with `supported_python_versions`.